hw/dv/dpi/usbdpi/usb_transfer.c

```diff
@@ -123,6 +150,19 @@ void transfer_data_end(usbdpi_transfer_t *transfer, uint8_t *dp) {
   transfer->num_bytes = (dp + 2) - transfer->data;
 }
 
+// Append some data to a transfer description
+bool transfer_append(usbdpi_transfer_t *transfer, const uint8_t *dp, size_t n) {
+  unsigned num_bytes = transfer->num_bytes;
+
+  // Check that we still have space to append this data
+  if (sizeof(transfer->data) - num_bytes < n) {
```

Comment:
The test would need to be > with the expression that you suggest, since the 'true' clause is the failure case in which there is insufficient space for the data.

The reason that I express it in the manner sizeof() - num_bytes may appear pedantic; sizeof() is a 'relatively small' constant value, and the 'num_bytes' field is a member of the 'usbdpi_transfer_t' structure, and thus controlled by the transfer_() code.

If we suppose the precondition '0 <= num_bytes <= sizeof()' is met, ie. the structure is in a valid state, the code works for all values of 'n' and the structure is left in a valid, defined state...'n' is a 32-bit 'size_t' quantity in practice, which means that if addition is used ('num_bytes + n') overflow could cause the code to go awry in the event that 'n' is a very large unsigned value.